step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/AssertHandler.java

```diff
@@ -134,8 +109,134 @@ protected void execute_(AssertReportNode node, Assert artefact) {
+			valueResolvingResult.actualResolved = true;
+		}
+		valueResolvingResult.type = ValueType.JSON_PATH;
+		return valueResolvingResult;
+	}
+
+	private ValueResolvingResult resolveSimpleValue(JsonObject outputJson, String key) {
+		ValueResolvingResult result = new ValueResolvingResult();
+		JsonValue jsonValue = outputJson.get(key);
+		if (jsonValue == null) {
+			// the attribute is missing (but we mark the value as resolved because some operators like 'notNull' support the missing values as nulls)
+			result.actual = null;
+			result.actualResolved = true;
+		} else if (jsonValue.getValueType() == JsonValue.ValueType.NULL) {
+			result.actual = null;
+			result.actualResolved = true;
+		} else if (jsonValue.getValueType() == JsonValue.ValueType.STRING) {
+			result.actual = outputJson.getString(key);
+			result.actualResolved = true;
+		} else if (jsonValue.getValueType() == JsonValue.ValueType.NUMBER) {
+			result.actual = outputJson.getJsonNumber(key).numberValue();
+			result.actualResolved = true;
+		} else if (jsonValue.getValueType() == JsonValue.ValueType.FALSE || jsonValue.getValueType() == JsonValue.ValueType.TRUE) {
+			result.actual = outputJson.getBoolean(key);
+			result.actualResolved = true;
+		} else {
+			result.message = "Type of " + key + " (" + jsonValue.getValueType() + ") is not supported";
+			result.actualResolved = false;
+		}
+
+		result.type = ValueType.SIMPLE;
+		return result;
+	}
+
+	private AssertResult applyOperator(String key, ValueResolvingResult valueResolvingResult, Object expectedValue, boolean negate, AssertOperator operator) {
+		Object actual = valueResolvingResult.actual;
+		ValueType type = valueResolvingResult.type;
+
+		AssertOperatorHandler handler = getOperatorHandler(operator);
+
+		if (!handler.isActualValueSupported(actual)) {
+			String message;
+
+			if (valueResolvingResult.message != null && !valueResolvingResult.message.isEmpty()) {
+				// in some cases (like in case of missing attributes) the error message is already prepared during value resolving
+				message = valueResolvingResult.message;
+			} else {
+
+				if (actual == null) {
+					// user-friendly message for null-value
+					message = "Unable to execute assertion. The keyword output doesn't contain the attribute '" + key + "'";
+				} else if (type == ValueType.JSON_PATH) {
+					// json path value
+					message = "The json path '" + key + "' returns an object of type "
+							+ actual.getClass().getSimpleName() + " which is not supported for operator " + operator.name();
+				} else {
+					// simple value
+					message = "Type of " + key + " ("
+							+ actual.getClass().getSimpleName()
+							+ ") is not supported for operator " + operator;
+				}
+			}
+			return createFailedAssertResult(message);
+		}
+
+		if (!handler.isExpectedValueSupported(expectedValue)) {
+			String message;
+			if (expectedValue == null || (expectedValue instanceof String && ((String) expectedValue).isEmpty())) {

```

Comment:
@iegorov777 This path is not covered by JUnit tests. I would recommend to had a simple JUnit test that covers non supported expected values

Reply:
In last commit I've added some simple test cases for this path.

Reply:
@jeromecomte 
> With following error "'keyInt' expected to be equal to '777' but was '777'"
This bug is also fixed under my last commit. Also I've changed the type of `expected` field from String to Object in AssertReportNode, because now we support various types of expected value.

Reply:
@jeromecomte 
I've re-checked the coverage. Now all these parts of code are covered either by AssertHandlerTest or by GreaterThanOperatorHandlerTest, EqualsOperatorHandlerTest etc.